core/client/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -404,4 +411,64 @@ private void updateBlockInStream(long blockId) throws IOException {
       mShouldCacheCurrentBlock = mAlluxioStorageType.isStore();
     }
   }
+
+  /**
+   * Seeks to a file position. Blocks are not cached unless they are fully read. This is only called
+   * by {@link FileInStream#seek}.
+   *
+   * @param pos The position to seek to. It is guaranteed to be valid (pos >= 0 && pos != mPos &&
+   *            pos <= mFileLength)
+   * @throws IOException if the seek fails due to an error accessing the stream at the position
+   */
+  private void seekWithoutCachingIncompleteBlocks(long pos) throws IOException {
```

Comment:
This can probably just be called `seekInternal()`, and the other version can be `seekInternalWithCaching()`? What do you think?